### PR TITLE
Handle zero tanks for sale

### LINKED ISSUE #484: Error on loading marketplace tanks when there are no tanks
**Describe the bug**
When you load into the tank listings and there are no tanks, there is an error message the reads "A valid MongoID is required". This is probably due to not checking if there are no tanks before getting the tanks.

**To Reproduce**
1. Go to marketplace tanks listings view when there are no tanks.
2. View error toast.

**Expected behavior**
Should have no message print.

**Screenshots**
image

**Additional context**
Also, the section should probably say "No Tanks in the Marketplace!"

### LINKED ISSUE #496: Make the marketplace say no tanks for sale
**Describe the bug**
When there are no tanks for sale the marketplace the marketplace should say there are no tanks for sale

**To Reproduce**
1. Go to 'Marketplace.'
2. Click on 'Tanks.'
3. See that it does not say no tanks for sale

**Expected behavior**

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by JorgeVidal29**
**Description**
We were having a bug when the api was called with one tank in the marketplace. That has been handled. Also added an edge case for when we have nothing in the marketplace.

**Resolves These Issues**
Resolves #496 
Resolves #484 

**Type of change**
Check options that are relevant.

- [x] Bug fix (fixes a bug issue)
- [ ] New feature (adds functionality)
- [ ] This fix or feature will cause existing functionality to not work as expected. (Please elaborate in Description.)

**Steps to Verify Functionality**
Make a bulleted/numbered list of steps to verify this feature/bugfix. Include screenshots if necessary.

**Final Checklist:**
Go down the list and check them off.

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have performed a self-review of my own code
- [x] My changes pass Flow, build without errors and (if applicable) pass Jest tests.
- [ ] This change requires a update in the design document (if applicable)

### CODE REVIEW

**Review by Derplime — APPROVED** [review]
Wooo

File: src/backend/routes/tankRoutes.js [hunk 1]
```diff
@@ -118,7 +118,7 @@ router.get('/getBotTanks', tankController.getBotTanks);
 // Takes an array of tank ids
 // returns the tank objects
 router.get('/getTanksById', [
-	check('array', 'An array of MongoIds is required.').isArray().isMongoId()
+	check('array', 'A JSON array of MongoIds is required.').isJSON()
 ], tankController.getTankById);
```
Comment (SecondThread) [inline src/backend/routes/tankRoutes.js]: I didn't notice this the first time, but getTanksById maps to getTankById? Sounds like a confusing name unless there is something going on here that I'm not understanding...

Comment (Derplime) [reply src/backend/routes/tankRoutes.js]: It's just a reference to the function in tankController. If we want them to match it's just a matter of matching the names in both files

Comment (Derplime) [reply src/backend/routes/tankRoutes.js]: In the grand scheme of things it doesn't affect functionality

**Pull request merged**